### DISCUSSION TITLE
Issue with DHT11 in a Sonoff 4CH Pro R3

### DISCUSSION THREAD

**Discussion by Kliwer** [arendst/Tasmota#15450] [Q&A] [answered]
Hi.
I flashed my Sonoff 4CH Pro with Tasmota. It works but I have two problems:
1. I have one DHT11 attached to sonoff but in Tasmota configuration I have to configure at least two of them to get a reading.
![image](https://user-images.githubusercontent.com/11368153/164996763-585cb549-33ea-44d0-8815-06fc5dadfdae.png)
![image](https://user-images.githubusercontent.com/11368153/164996781-bc945520-5581-4013-9271-60e91b3f1071.png)
If I leave only GPIO3, all I get is null.
2. Tasmota is spamming MQTT every second:
22:10:57.401 MQT: domoticz/in = {"idx":13,"nvalue":0,"svalue":"","Battery":100,"RSSI":7}
22:10:57.407 MQT: domoticz/in = {"idx":12,"nvalue":0,"svalue":"","Battery":100,"RSSI":7}
22:10:57.413 MQT: domoticz/in = {"idx":14,"nvalue":0,"svalue":"","Battery":100,"RSSI":7}
22:10:58.421 MQT: domoticz/in = {"idx":13,"nvalue":0,"svalue":"","Battery":100,"RSSI":7}
22:10:58.427 MQT: domoticz/in = {"idx":12,"nvalue":0,"svalue":"","Battery":100,"RSSI":7}
22:10:58.433 MQT: domoticz/in = {"idx":14,"nvalue":0,"svalue":"","Battery":100,"RSSI":7}
Is there a way to disable this?

**Comment by laurentppol** [comment]
Hi,
have You configured Domoticz? What for?
Did You try putting both DHTs on the same pin?
I don't know this type of sensor, but if it is 1Wire compliant it would be enough to have 1 bus and all devices on it.

**Reply by Kliwer** [reply]
Yes, Domoticz is configured. 4 light switches + temperature and humidity. 
I have only ONE DHT11, but if there is only one configured in Tasmota it's not working.

**Comment by Jason2866** [comment]
Tasmota does not support DHT sensors on more Gpio pins. If you get no readings your DHT sensor is out of specs. Anyways DHT 11 is a very very bad temp sensor.

**Reply by Kliwer** [reply]
Can't people read?
I have ONE sensor. ONE! On GPIO3 (does not really matter, I tried every GPIO)
But if I select in Tasmota that I have ONE sesnor then I get no reading.
If I select more, then the reading from sensor is there.

**Reply by Jason2866** [reply]
I can read ;-) If the DHT11 is not working, it is a bad one. Trash it and get a real sensor.

**Reply by Kliwer** [reply]
It is working...

**Reply by sfromis** [reply]
My experience with DHT11 is, to put it kindly, random. It may work, sometimes, and sometimes craps out. I have much better experience with I2C senors instead of that finicky toy.

**Reply by Kliwer** [reply]
I have no problem with DHT11. It works fine for me. 
The problem is in Tasmota but I see everyone is focusing only on DHT11.

One more time:
If in Tasmota configuration I select DHT on ONE GPIO pin (doesn't matter which one) I get null values.
If in Tasmota configuration I select DHT on TWO GPIO pins (only ONE is connected) then on first one I get null (correct as it is not connected), on second I get proper values.

**Reply by sfromis** [reply]
Your tale is clearly about a finicky DHT11, not an unproblematic one. Potentially, Tasmota could have adjusted timings to deal with your particular one, but then it might work worse for others.

**Reply by Kliwer** [reply]
Yeah, right. For the past two months configured like I wrote earlier works fine. But sure, DHT11 is faulty and probably the reason for MQTT spamming too? EOT.

**Reply by sfromis** [reply]
MQTT is another issue no one has tried giving an answer to, maybe due to lack of details from your side.

I very deliberately did not use the word "faulty", but "finicky". 

Just tested a bit more. Sometimes it works on those pins, sometimes not. Only fun if enjoying that type of "games".... 

And I very deliberately suggested that it might be a tweakable timing issue, which would depend on the Tasmota driver. It's just that we have so damn many issues with those one-wire sensors, with it being a moving target when it works, and when it does not.

**Comment by Kliwer** [comment, marked as answer]
As it seems Tasmota is bugless there is no point in wating anymore for answers. EOT

**Reply by Jason2866** [reply]
Using serial port gpios is problematic for this type of sensor. We tried using other gpios and it does work.
